utils/utils.go

```diff
@@ -147,3 +149,42 @@ func EncryptionText(enum int) string {
 
 	return "unknown"
 }
+
+func isUrl(str string) bool {
+	if _, err := url.Parse(str); err != nil {
+		return false
+	}
+
+	return strings.Contains(str, "://")
+}
+
+func PathToUrl(u string) (*url.URL, error) {
+	url := &url.URL{}
+	// Check if the given string starts with a protocol scheme. Prepend the file
+	// scheme in case none is provided
+	if !isUrl(u) {
+		url.Scheme = "file"
+		url.Path = u
+	} else {
+		// u = url.QueryEscape(u)
+		var err error
+		url, err = url.Parse(u)
+		if err != nil {
```

Comment:
You could just use "return url, err"

Reply:
No, we only want to return in case of an error here. Otherwise we'll proceed with the parsed url.

Reply:
Oops! Thank you & sorry.